ssh_key_rotator/custom_keygen.py

```diff
@@ -41,15 +40,12 @@ def generate_private_public_key_in_file(
     private_key_path = f"{private_key_dir}/{private_key_name}"
     public_key_path = f"{public_key_dir}/{public_key_name}"
 
-    private_key = key.export_key()
-
     with open(private_key_path, "wb") as private_out:
         private_out.write(private_key)
 
     shutil.chown(private_key_path, user=user, group=-1)
     os.chmod(private_key_path, 0o600)
```

Comment:
it's usually better to create the file with the right permissions from the outside (the way we have here it's a bit of a security issue, since the original file will be unprotected for some time)

if you look in the documentation for [`open`](https://docs.python.org/3/library/functions.html#open) you will see it takes an `opener` function - that's where you can invoke `chmod` to create the file with right permission from the outset